master/internal/trials/api_trials.go

```diff
@@ -338,3 +345,55 @@ func (a *TrialsAPIServer) DeleteTrialsCollection(
 
 	return &apiv1.DeleteTrialsCollectionResponse{}, nil
 }
+
+// ValidationMetricsSeries returns a time-series of the specified validation metric in the specified
+// trial.
+func ValidationMetricsSeries(trialID int32, startTime time.Time,
+	metricName string,
+	startBatches int, endBatches int, xAxisMetricLabels []string,
+	maxDatapoints int, rangeType apiv1.RangeType,
+	integerRange *commonv1.Int32FieldFilter,
+	timeRange *commonv1.TimestampFieldFilter) (
+	metricMeasurements []db.MetricMeasurements, err error,
+) {
+	var queryColumn, orderColumn string
+	measurements := []db.MetricMeasurements{}
+	subq := db.Bun().NewSelect().TableExpr("validations").
```

Comment:
suggestion: for reproducibility and our general testing sanity, we should use a fixed random seed for these queries. for example, if we fetch the metrics for a finished trial, the API result should be the same every time.
probably need to use `setseed` for this: https://www.postgresql.org/docs/current/sql-set.html